### PR TITLE
Add ban/unban command

### LINKED ISSUE #1073: Add ban and unban commands
Currently, users have to do "/mode username +b" and "/mode username -b". Many irc clients have /ban and /unban as links to that.

Also, [irc-fw just added some nice new commands for these (and banlist)](https://github.com/kiwiirc/irc-framework/pull/89/files) so we should be able to simplify the /mode command quite a bit (it's quite ... interesting).

### PR DESCRIPTION
**Pull request by YaManicKill**
Fixes #1073

### CODE REVIEW

File: src/plugins/irc-events/banlist.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ module.exports = function(irc, network) {
 	irc.on("banlist", function(banlist) {
 		const channel = banlist.channel;
 		const bans = banlist.bans;
-		if (!bans) {
+		if (!bans || bans.length === 0) {
```
Comment (YaManicKill) [inline src/plugins/irc-events/banlist.js]: It appears that we now get through an empty array rather than undefined when there are no bans. So this needs updated. This was added in [irc-framwork](https://github.com/kiwiirc/irc-framework/commit/1cc8892f6bf5414e677f0d37daa5984b77b58374).

File: src/plugins/inputs/ban.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+				text: `Usage: /${cmd} <nick>`
+			}));
+
+			return;
+		}
+	}
+
+	switch (cmd) {
+	case "ban":
+		network.irc.ban(chan.name, args[0]);
```
Comment (xPaw) [inline src/plugins/inputs/ban.js]: Will this ban just the nick? I think most channels do hostname bans.

Comment (YaManicKill) [inline src/plugins/inputs/ban.js]: It'll bad whatever you pass it, so it'll either be nick or hostmask depending on what you pass it.

"/ban test" will ban the user
"/ban *!*@test.com" will ban that hostmask

Comment (MiniDigger) [inline src/plugins/inputs/ban.js]: typing out hostnames is meh, what about a /banh command to ban the hostname of a user?

Comment (YaManicKill) [inline src/plugins/inputs/ban.js]: I think that's out of scope of this PR. I would expect that ban doesn't automatically assume I want to ban the hostname or anything like that, and just dumbly send it on. So if we want a new command, that's fine, but that's not part of this PR.

Comment (xPaw) [inline src/plugins/inputs/ban.js]: Bans only apply to hostmasks though, do all server support simply sending a ban for a nickname?

Comment (YaManicKill) [inline src/plugins/inputs/ban.js]: I mean, this isn't any new functionality, because if we send a "/mode +b username" then it's exactly the same. All this does is basically symlink "/ban" to "/mode +b" with some checks that you are sending something. 

[The spec is vague on it.](https://tools.ietf.org/html/rfc1459#section-4.2.3)

File: client/index.html [hunk 3]
```diff
@@ -556,6 +556,15 @@ <h1 class="title">Help</h1>
 
 						<div class="help-item">
 							<div class="subject">
+								<code>/ban user</code>
```
Comment (astorije) [inline client/index.html]: All other helpers are `/foo nick`, is this one different on purpose? If it's to highlight that not only nicks can be passed, then I suggest `/ban target` and then in the description `<code>target</code> can be a nickname, or a...` (with a better wording than what my 5-yo English can come up with lol).

File: src/plugins/irc-events/banlist.js (hunk 1 above)
Comment (astorije) [inline src/plugins/irc-events/banlist.js]: Kind of a breaking change on their end if you ask me, but oh well. Good catch!

Comment (YaManicKill) [inline src/plugins/irc-events/banlist.js]: Yeah, probably, but too late now :-P

**Comment by YaManicKill** [comment]
> (You do have a passion for banning people, don't you? :D)

What can I say? I don't like people :-P

File: client/index.html (hunk 3 above)
Comment (YaManicKill) [inline client/index.html]: Nah, accidental. I'll add it just as "nick" but point out it can be a hostmask,

Comment (YaManicKill) [inline client/index.html]: Done.

File: client/index.html [hunk 4]
```diff
@@ -801,6 +811,15 @@ <h1 class="title">Help</h1>
 
 						<div class="help-item">
 							<div class="subject">
+								<code>/unban user</code>
```
Comment (astorije) [inline client/index.html]: Aaaand the other one? :D

Comment (YaManicKill) [inline client/index.html]: https://media.giphy.com/media/Kh296M49Zr1Sg/giphy.gif

Goodness, it's clearly too early.

Comment (astorije) [inline client/index.html]: And I thought I was the one who hasn't slept all night :P
